### PR TITLE
feat(rc): move ix-collectd and ix-system to pool.import_on_boot

### PR DESCRIPTION
**Pull request by william-gr**

### CODE REVIEW

File: src/middlewared/middlewared/plugins/pool.py [hunk 1]
```diff
@@ -2379,6 +2379,14 @@ def import_on_boot(self, job):
                     'aclmode': {'value': 'restricted'},
                 }})
 
+        try:
+            # Now that pools have been imported we are ready to configure system dataset
+            # and collectd which may depend on them.
+            self.middleware.call('etc.generate', 'system_dataset')
+            self.middleware.call('etc.generate', 'collectd')
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/pool.py]: Is this intended to be skipped if `system_dataset` fails?

Comment (william-gr) [reply src/middlewared/middlewared/plugins/pool.py]: Yes, there is no point in raising the exception and not continue on remaining tasks

Comment (themylogin) [reply src/middlewared/middlewared/plugins/pool.py]: I mean, skip setting up collectd if system_dataset setup failed

Comment (themylogin) [reply src/middlewared/middlewared/plugins/pool.py]: We probably need to prevent starting collectd if it is not set up

Comment (william-gr) [reply src/middlewared/middlewared/plugins/pool.py]: If system dataset fails to the point of generating an exception we are probably already dead in the water and the system will malfunction

**Pull request merged**